### PR TITLE
LibGUI: Reduce datetime instantiation in Calendar::update_tiles

### PR DESCRIPTION
**Pull request by adamantike**
Currently, navigating through different years in the Year view of the Calendar app or the taskbar is very slow.
Profiling results show that almost all the time is spent in `Calendar::update_tiles`, and specifically, in `DateTime::create` and `DateTime::set_time`. Performance can improve substantially if the `TZ` environment variable is set [0], but we can improve the current code to perform better nevertheless :^)

This diff includes the following changes:

* Take advantage of `DateTime::create` internally using mktime, which corrects out of bound components, to create valid dates. That way, we can considerably reduce the amount of created intermediate datetimes, and avoid modifying them with `set_time`, by using some math.
* Only retrieve the current `DateTime::now()` once.

#### How to test

* Open the Calendar app, switch to Year view, and change between different years with the left/right arrows. Notice CPU usage, and how much it takes to render a different year.
* Click on the current hour in the taskbar, to see the calendar, and click on the month label to change to Year view. Change between different years, and notice how much it takes to load a new year.

[0] https://blog.packagecloud.io/set-environment-variable-save-thousands-of-system-calls/

### CODE REVIEW

**Comment by adamantike** [comment]
cc'ing @thankyouverycool, who seems to be the main author of the code this PR refactors.

File: Userland/Libraries/LibGUI/Calendar.cpp [hunk 1]
```diff
@@ -264,38 +265,26 @@ void Calendar::update_tiles(unsigned view_year, unsigned view_month)
...
 
-    unsigned months;
-    mode() == Month ? months = 1 : months = 12;
+    auto now = Core::DateTime::now();
+    unsigned months = mode() == Month ? 1 : 12;
     for (unsigned i = 0; i < months; i++) {
         if (mode() == Year)
             view_month = i + 1;
+
+        unsigned first_day_of_week_for_month = ::day_of_week(view_year, view_month, 1);
```
Comment (thankyouverycool) [inline Userland/Libraries/LibGUI/Calendar.cpp:274]: No need to scope AK here

**Comment by thankyouverycool** [comment]
Nice improvement, much more readable :+1:

File: Userland/Libraries/LibGUI/Calendar.cpp (hunk 1 above)
Comment (adamantike) [reply Userland/Libraries/LibGUI/Calendar.cpp:274]: I'm trying to reduce the amount of DateTime instantiations as much as possible. If `::day_of_week` isn't something we want to use directly, would you prefer this to still avoid a new instantiation?

```cpp
unsigned first_day_of_week_for_month = (first_day_of_previous_month.weekday() + first_day_of_previous_month.days_in_month()) % 7;
```

Comment (thankyouverycool) [reply Userland/Libraries/LibGUI/Calendar.cpp:274]: I mean you can remove the `::`
AK is global and no ambiguity here

Comment (adamantike) [reply Userland/Libraries/LibGUI/Calendar.cpp]: Resolved!

**Comment by adamantike** [comment]
Superseded by #13186, which provides better performance and less DateTime instantiations.

**Pull request closed without merging**